### PR TITLE
Facade should ignore burn damage reduction

### PR DESCRIPTION
**Pull request by mattrossdev**
Facade should ignore the burn damage reduction of physical moves: https://bulbapedia.bulbagarden.net/wiki/Facade_(move)

I didn't add a move flag for this because it only applies to this move, but I can rewrite it that way if that's preferred. 

Testing: 
- non facade contact moves do half damage when burned 
- facade does 2x damage if burned

### CODE REVIEW

**Comment by happinyz** [comment]
Just curious, do you happen to know if Guts also has the same bug?

**Comment by mattrossdev** [comment]
> Just curious, do you happen to know if Guts also has the same bug?

Just looking at the code I don't think so but haven't tested

File: src/field/pokemon.ts [hunk 1]
```diff
@@ -1403,7 +1403,7 @@ export default abstract class Pokemon extends Phaser.GameObjects.Container {
 
           if (!isTypeImmune) {
             damage.value = Math.ceil(((((2 * source.level / 5 + 2) * power.value * sourceAtk.value / targetDef.value) / 50) + 2) * stabMultiplier.value * typeMultiplier.value * arenaAttackTypeMultiplier.value * screenMultiplier.value * ((this.scene.randBattleSeedInt(15) + 85) / 100) * criticalMultiplier.value);
-            if (isPhysical && source.status && source.status.effect === StatusEffect.BURN) {
+            if (isPhysical && source.status && source.status.effect === StatusEffect.BURN && move.id !== Moves.FACADE) {
```
Comment (bennybroseph) [inline src/field/pokemon.ts:1406]: i think we can move this check to the front of the if statement, no? There's no reason to check anything else if you're using Facade.

**Review by bennybroseph — CHANGES_REQUESTED** [review]

File: src/field/pokemon.ts (hunk 1 above)
Comment (mattrossdev) [reply src/field/pokemon.ts]: Updated

**Comment by bennybroseph** [comment]
Would like to make this an Attribute instead of hard-coding only Facade to work like this. Abilities are already doing so right below this code, so let's be consistent.

File: src/field/pokemon.ts [hunk 2]
```diff
@@ -1403,7 +1403,7 @@ export default abstract class Pokemon extends Phaser.GameObjects.Container {
 
           if (!isTypeImmune) {
             damage.value = Math.ceil(((((2 * source.level / 5 + 2) * power.value * sourceAtk.value / targetDef.value) / 50) + 2) * stabMultiplier.value * typeMultiplier.value * arenaAttackTypeMultiplier.value * screenMultiplier.value * ((this.scene.randBattleSeedInt(15) + 85) / 100) * criticalMultiplier.value);
-            if (isPhysical && source.status && source.status.effect === StatusEffect.BURN) {
+            if (move.id !== Moves.FACADE && isPhysical && source.status && source.status.effect === StatusEffect.BURN) {
               const burnDamageReductionCancelled = new Utils.BooleanHolder(false);
               applyAbAttrs(BypassBurnDamageReductionAbAttr, source, burnDamageReductionCancelled);
               if (!burnDamageReductionCancelled.value)
```
Comment (bennybroseph) [inline src/field/pokemon.ts:1409]: referring to these few lines here

**Review by bennybroseph — CHANGES_REQUESTED** [review]

File: src/data/move.ts [hunk 3]
```diff
@@ -1317,6 +1317,14 @@ export class BypassSleepAttr extends MoveAttr {
   }
 }
 
+export class BypassBurnDamageReductionAttr extends MoveAttr {
+  apply(user: Pokemon, target: Pokemon, move: Move, args: any[]): boolean {
+    (args[0] as Utils.BooleanHolder).value = true;
+
+    return true
```
Comment (bennybroseph) [inline src/data/move.ts:1324]: need a semi-colon here

File: src/data/move.ts [hunk 4]
```diff
@@ -1317,6 +1317,14 @@ export class BypassSleepAttr extends MoveAttr {
...
 }
 
+export class BypassBurnDamageReductionAttr extends MoveAttr {
+  apply(user: Pokemon, target: Pokemon, move: Move, args: any[]): boolean {
+    (args[0] as Utils.BooleanHolder).value = true;
+
+    return true 
+  }
+}
+
```
Comment (bennybroseph) [inline src/data/move.ts:1327]: Would like to get TSDoc documentation for this class/function

**Review by bennybroseph — CHANGES_REQUESTED** [review]

**Review by bennybroseph — APPROVED** [review]

**Pull request merged**